### ISSUE TITLE
Hall of Shame

### ISSUE THREAD

**Issue by zhanghai** [zhanghai/MaterialFiles#291]
Pleaes help reporting these copycat apps on Google Play. I believe they are violating the [Repetitive Content](https://play.google.com/intl/en_us/about/spam-min-functionality/#!?zippy_activeEl=repetitive-content#repetitive-content) policy inside Play Developer Policy.

- [File Manager by Appbox Apps (com.appbox.filemanager)](https://play.google.com/store/apps/details?id=com.appbox.filemanager): Closed source, contains ads.

**Comment by GajenderY** [comment]
Submitted.

**Comment by yarikvv** [comment]
Submitted

**Comment by thgoebel** [comment]
Also, I think they violate your license (GPLv3) by not naming you as the original source and not open sourcing their modifications of the source code.

**Comment by gprst** [comment]
It seems that a request to "remove content from Google" regarding "copyright infringement" must be done by the author themselves.

I submitted a request, but I think the app would be more likely to be removed from the Play Store if it had legal complaints.

**Comment by ssssamstrup** [comment]
Please advise to vote the app down in the post above. Submitted and voted the app 1-star with a comment telling about the copying. This way people will see it in the Play store.

**Comment by J-Jamet** [comment]
You must also install the application before putting 1 * otherwise the google algorithm considers the comment as spam.
Uninstall this pale copy right after obviously.

**Comment by abrahamxf** [comment]
发现了新的抄袭者！
https://play.google.com/store/apps/details?id=x.gen.android.files
以及它的高级版
https://play.google.com/store/apps/details?id=x.genpro.android.files

**Comment by PrSunflower** [comment]
This is really a shame and very disrespectful 😡, I've flagged the clones as inappropriate.
  
> Also, I think they violate your license (GPLv3) by not naming you as the original author and not open sourcing their modifications of the source code. (anyway, reported)

I was thinking about that too. Legally speaking, the GPL licences **oblige** publishing copies under the same license, don't they?

_Side note: Sorry for the notification above._ I copy/pasted the link to this issue in another completely different repository and I didn't know GitHub was going to mention it. I removed the link to this issue but the notification doesn't disappear, it can be deleted if you want.

**Comment by zhanghai** [comment]
Yes I believe they are not compliant with GPL, but IIRC Google Play asked me for a lawyer if I were to submit a copyright infringement form 😅

I won't be happy with people re-publishing this app mostly as-is, but with ads/paid version, even if they open source their app and comply with GPL anyway 🙃

**Comment by PrSunflower** [comment]
@zhanghai 
> Yes I believe they are not compliant with GPL, but IIRC Google Play asked me for a lawyer if I were to submit a copyright infringement form 😅

Oh no, having to work with a lawyer for a simple matter like this... 😕

If you're willing to spend some time maybe you could [contact GNU](https://www.gnu.org/licenses/gpl-violation.html) about this. They have instructions dedicated to "violations of the GNU Licenses" so I guess some developers already have had this issue.


> I won't be happy with people re-publishing this app mostly as-is, but with ads/paid version, even if they open source their app and comply with GPL anyway 🙃

And yes I totally understand this. If you publish a free/libre app the purpose is for people to use it and maybe even improve it if they have good ideas, not copying everything from the ground and adding malware in the code. 😠

**Comment by gotmax23** [comment]
It looks like the second one `File Manager Pro 2020 — File Manager` was taken down, but the other two are still up.

This seems to be a problem with a couple other FOSS apps as well. It's unfortunate that Google doesn't do anything about it.

**Comment by woodvale** [comment]
Found this one in Play Store [File Explorer FTP Server](https://play.google.com/store/apps/details?id=com.corproxy.files) by [Corproxy Innovation](https://play.google.com/store/apps/dev?id=5035150003288626185)

No ads but still a GPL violation

**Comment by shuvashish76** [comment]
https://support.google.com/legal/troubleshooter/1114905

![Screenshot_2021-08-03-19-18-00](https://user-images.githubusercontent.com/48226312/128026697-2ada7b95-b574-44ed-9e72-698a57c07463.jpeg)

According to Google they only accept copyright complaints from content owners or someone officially authorized to act on their behalf. We users cant report. @zhanghai Please report those apps from the above link & later mail them with side by side screenshots our original app & those apps.

Thankyou🙏🏿

**Comment by zhanghai** [comment]
@shuvashish76 It eventually leads me to a DMCA page that warns me I might need a lawyer and may be involved in expansive litigations. I would prefer avoiding that legal risk for just a side project of mine 🙃

![image](https://user-images.githubusercontent.com/4469895/128067488-a357af3e-48bd-4b9d-8507-96d13593e205.png)

**Comment by watson96** [comment]
True. It's futile effort to put on something that isn't a mainstream source of your earnings. Yours is open-source and they won't be able to act on your content anyways as it will be clearly seen that they're the ones who plagiarised your work were they made to reveal their source.

**Comment by ptalmeida** [comment]
Google is infamous for ignoring problems (scams, copycats, etc.) in it's playstore unfortunately.

**Comment by dariusity** [comment]
Submitted, can I ask you something quickly? Isit possible to add applications with some basic features? And isit possible for me to adjust the height of the columns? The texts seem too far apart and wasting screen pixels. Cheers.

**Comment by ashjas** [comment]
Reported all three clones.

**Comment by VA1DER** [comment]
> It eventually leads me to a DMCA page that warns me I might need a lawyer

They are just trying to scare people from making false claims that they own the copyright.  You clearly own the copyright and are entitled to make the claims you do.  I encourage you to continue with the takedown request.

**Comment by AlbertEienstein** [comment]
> Google is infamous for ignoring problems (scams, copycats, etc.) in its playstore unfortunately.

But zhanghai is a Google Employee, so Google shouldn't ignore copycats of his own Project.

**Comment by lipici** [comment]
Still there, as long as developer dosent open ticket to google its just wasted time from us.

**Comment by Haleymiranda79** [comment]
The internet hates me.

**Comment by zhanghai** [comment]
> But zhanghai is a Google Employee, so Google shouldn't ignore copycats of his own Project.

That's just your assumption. I did and that's not the case.

**Comment by z8512** [comment]
Submitted.

**Comment by J-Jamet** [comment]
What you say is just gratuitous hate.

You really want to spit on the work of someone who gives his time and energy to the community just because he works for a living?

The philosophy of a company does not necessarily reflect that of an employee, even all branches of a single company may not even have the same ambitions.
We have to stay factual, we have to remember the AOSP, and think that we are in a capitalist world, whether we like it or not.

Personally, I obviously thank the work of @zhanghai who is exemplary on this project and denounce copycats who do not respect that.

**Comment by zhanghai** [comment]
Thank you @J-Jamet for speaking up for me. I didn't intend to reply but I feel I need to put a stop to this drama.

I work on Android itself at Google and my code changes are released in AOSP. So if you (@therealbluepandabear) don't like or have no respect for people who work on AOSP (the vast majority of them being at Google) I don't see why you are (presumably) still using AOSP or any Android apps including this one.

I believe the fact that I developed this FOSS project means I am a FOSS developer, regardless of what I do as my day job, your imagination about me, or your personal definition of a "true" FOSS developer. And I'm not happy with someone openly and repeatedly denying me of that.

If you have some opinions towards me personally, this is NOT the right issue or forum to talk about it. If you feel strongly about voicing that opinion you can send that as an email to me so that I can ignore it without disturbing anyone else. But this issue is about others violating the license of this project, and this repository is about the project itself instead of me, so please refrain from voicing your personal opinion on me and distrubing people who subscribed to this or want to actually discuss the subject here. If you keep doing that, I'll have to block you from this forum.

(I see you are actively editing your comment, so please also replace your images with links or just the plain text of those quotes to reduce the distraction it brought to this issue, or I'll do that for you.)

**Comment by zalnaRs** [comment]
There is a wearos application which is basically material files. https://play.google.com/store/apps/details?id=com.corproxy.files

**Comment by daiaji** [comment]
The GPL agreement is recognized by law in most areas, but Google's attitude is surprising.

**Comment by ch3thanhs** [comment]
Found another one - https://play.google.com/store/apps/details?id=me.geosoftech.android

**Comment by JamesClarke7283** [comment]
> Please advise to vote the app down in the post above. Submitted and voted the app 1-star with a comment telling about the copying. This way people will see it in the Play store.

It appears you have to pay to leave a review (at least for me i cant leave a comment on them so i assume).

(Although i already filed the reporta so that might block it, idk).

Seems like we need a all in one solution to auto detect and streamline the reporting process (semi-semi-automate).

Many of these apps fall into the same buckets.

Can also help remove malware.

I am sure Google will eventually use their ML technology(and i dont just mean an LLM, purpose built system) to help clean the malware off at least.

I realise they don't have the incentive for some stuff. But stuff like this dilutes their products effectiveness (think of the amount of garbedge the Google employees have to sift through, that costs lots or money).

Google could also autogenerate parts of the "Data Safety" section by default (probably is a tool for it). I prefered it when they showed the permissions on the listing, now i need to use "Exodus Privacy" or something to check. (Logically Why not make both available? One is not a substitute for the other, my question is rhetorical of course, i know some of the reasons why).

The nature is such that some of the data policy could be calculated with a near zero error rate.

Like knowing  *some of Who data is shared with (at least the direct part, because Android apps are Java Bytecode, one can deterministically know that certain AST patterns WILL generate a network request). Eeer on the side of causion for any patterns that arent certain.

Indirect could be an added detail.

I use F-Droid for 90%+ of my apps so it doesnt  impact me directly.

**Comment by sosasees** [comment]
you should strike these dead links 🎉

> - ~~[File Manager Advanced - File Explorer](https://play.google.com/store/apps/details?id=x.gen.android.files) by [X-Gen]()~~. Taken Down
> - ~~[File Explorer FTP Server](https://play.google.com/store/apps/details?id=com.corproxy.files) by [Corproxy Innovation](https://play.google.com/store/apps/dev?id=5035150003288626185): Closed source, 50,000+ installs~~ Taken Down
> - ~~[File Manager Pro](https://play.google.com/store/apps/details?id=me.geosoftech.android) by [Premium Quality Apps](https://play.google.com/store/apps/developer?id=Premium+Quality+Apps): Probably closed source (didn't purchase to check), paid app, 1,000+ installs~~ Taken Down
